src/super_gradients/common/decorators/code_save_decorator.py

```diff
@@ -0,0 +1,29 @@
+import inspect
+
+import os
+
+saved_codes = {}
```

Comment:
Not sure this is the best idea...

If we treat saved_codes as a global var then consecutive runs will result in weird loggings...